driver/pace/driver/driver.py

```diff
@@ -482,23 +642,15 @@ def _write_performance_json_output(self):
         )
 
     @dace_inhibitor
-    def _write_restart_files(self, restart_path="RESTART"):
-        self.restart.save_state_as_restart(
+    def cleanup(self):
+        logger.info("cleaning up driver")
+        self.config.restart_config.write_final_if_enabled(
             state=self.state,
-            comm=self.comm,
-            restart_path=restart_path,
-        )
-        self.restart.write_restart_config(
             comm=self.comm,
             time=self.time,
             driver_config=self.config,
-            restart_path=restart_path,
+            restart_path="RESTART",
         )
```

Comment:
It looks to me that some of these don't need to be part of dace call back, not sure if this changes the behavior of dace.